src/LostDogForm.js

```diff
@@ -0,0 +1,219 @@
+import React from "react";
+import AnnouncementData from "./AnnouncementData";
+import {connect} from "react-redux";
+
+class LostDogForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.lostDogRef = React.createRef();
+        this.state = {
+            name: '',
+            dogsname: '',
+            dogsbread: '',
+            location: '',
+            message: '',
+            phoneNumber: '',
+            email: '',
+            filename: '',
+            imgSource:'',
+            uploading: false
+        };
+
+        this.handleInputChange = this.handleInputChange.bind(this);
+    }
+
+
+    // https://stackoverflow.com/questions/9267899/arraybuffer-to-base64-encoded-string
+
+    arrayBufferToBase64( buffer ) {
```

Comment:
its okay to get code from stackoverlow, but you should make sure it fits with style guides of your project.